Adding HPA for eventing-webhook

matzew opened the pull request:
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

## Proposed Changes

- Set up HPA for eventing-webhook:
  - Scale Webhook based on CPU
  - Add initial PodDisruptionBudget

Similar to this PR on the serving component from September 2020: https://github.com/knative/serving/pull/9444


```release-note
Adding HorizontalPodAutoscaler and PodDisruptionBudget for the eventing webhook
```

devguyio commented:
/assign

devguyio commented:
- I understand serving have it, and the only reasoning I saw was to not have a SPOF. However, given that this is a control plane component, I'm not really sure how valuable is the replication based on CPU. I can imagine it might starve on memory more, I think @mattmoor had such an opinion? also, if we want to achieve high availability, shouldn't the min be 2? 
- I don't have a strong opinion, other than the question stated above. 
- @matzew I think it might be useful to have a release note? something like "The eventing webhook has now a horizontal pod autoscaler based on ......etc" 

lgtm if folks wanna have it in , I'll leave this for someone with a stronger opinion about it. @vaikas @grantr @lionelvillard

pierDipi commented on config/core/deployments/webhook-hpa.yaml:
```diff
@@ -0,0 +1,51 @@
+  minAvailable: 80%
+  selector:
+    matchLabels:
+      app: webhook
```
This should be `eventing-webhook`.

matzew commented:
Related I will be also working introducing `high-availability` field on the `KnativeEventing` spec, of the operator.

There you set the `replicas`.  This is more ground work, and adding a HPA/scaling for the stateless webhook (exactly like we have on serving)

pierDipi commented on config/core/deployments/webhook-hpa.yaml:
```diff
@@ -0,0 +1,51 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: autoscaling/v2beta1
```
Can't we use `v2beta2`?

matzew commented:
@pierDipi true: I updated to `v2beta2` and fixed the `eventing-webhook`

This matches now the HPAs we have already in for the MT Broker, which also goes for the `cpu` resource

devguyio commented:
lgtm

leaving it for @pierDipi

pierDipi reviewed (commented):
/lgtm

pierDipi commented:
/kind enhancement
